projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.ts

```diff
@@ -113,8 +113,9 @@ export class IgxChildGridRowComponent implements AfterViewInit, OnInit {
 
     public set data(value: any) {
         this._data = value;
-        if (this.hGrid) {
+        if (this.hGrid && !this.hGrid.dataSetByUser) {
             this.hGrid.data = this._data.childGridsData[this.layout.key];
+            this.hGrid.dataSetByUser = false;
```

Comment:
I'm not a fan of having to set this back to false any time the data is set internally. Maybe refactoring it a bit so that there's a setDataInternal method we can use for any internal operation that does not set this `dataSetByUser` flag?